src/bionics.cpp

```diff
@@ -1939,18 +1939,16 @@ int player::get_free_bionics_slots( const body_part bp ) const
 
 void player::add_bionic( const bionic_id &b )
 {
-    if( has_bionic( b ) ) {
+    if( has_bionic( b ) && !bionics[b].allow_duplicates ) {
         debugmsg( "Tried to install bionic %s that is already installed!", b.c_str() );
         return;
     }
 
     units::energy pow_up = bionics[b].capacity;
     mod_max_power_level( pow_up );
-    if( b == "bio_power_storage" || b == "bio_power_storage_mkII" ) {
+    if( pow_up > 0_J ) {
         add_msg_if_player( m_good, _( "Increased storage capacity by %i." ),
                            units::to_kilojoule( pow_up ) );
```

Comment:
The will not report if the max power wsa increased by an amount less than one joule.
Additionally, the message will show zero if the power was increased by any value less than 1 kilojoule

Reply:
Ok, thanks for letting me know. I'll look at the new menu code to see how they handle power display. I wasn't sure what to use for 0 in the new system.

Reply:
The message being wrong is technically not your problem to solve (you don't touch it), but it'd be nice if you did fix it.

Reply:
In fact, I might suggest (if there's no function to format the power like the menu/sidebar do), that you don't do that and then you or someone else makes that function and uses it in the proper places in a separate PR.

Reply:
Changed 0_J to 0_mJ, it looks like that's the smallest unit.